README.md

```diff
@@ -71,6 +71,144 @@ You can customize your own macros by modifying the `GSH_AGENT_MACROS` configurat
 The value should be a JSON object mapping macro names to their corresponding chat messages.
 See [Configuration](#configuration) for more details.
 
+#### Permission System
```

Comment:
I'm realizing the main README file is getting too long - do you mind creating a separate md file and link to it?